tests/test_request.py

```diff
@@ -249,3 +249,28 @@ def test_default_secure_false(self):
     def test_default_secure_true(self):
         request = Request(factory.get('/', secure=True))
         assert request.scheme == 'https'
+
+
+class TestWSGIRequestProxy(TestCase):
+    def test_access_inner_property(self):
+        wsgi_request = factory.get('/')
+
+        sentinel = object()
+        wsgi_request.__dict__['inner_property'] = sentinel
```

Comment:
It shouldn't be necessary to do reference the underlying `__dict__. Regular attribute setting should be sufficient.
```python
wsgi_request.inner_property = sentinel
```